lib/trino-parquet/src/main/java/io/trino/parquet/AbstractParquetDataSource.java

```diff
@@ -326,9 +326,9 @@ public Slice read()
 
             if (data == null) {
                 byte[] buffer = new byte[toIntExact(range.getLength())];
+                readerMemoryUsage.setBytes(buffer.length);
```

Comment:
nit: in practice it doesn't matter since memory usage is propagated in `ScanFilterOperator` in a lazy way anyway